File: week2/ts/src/repository/repository.ts
export interface Repository<T, Tid> {
    add: (item: T) => void;
    delete: (item: T) => void;
    update: (item: T, id: Tid) => void;
    findById: (id: Tid) => T;
    findAll: () => Iterable<T>;
    getAll: () => Array<T>;
}

File: week2/csharp/main/repository/Repository.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp.repository
{
    public interface IRepository<T, Tid>
    {
        void Add(T item);
        void Delete(T item);
        void Update(T item, Tid id);
        T FindById(Tid id);
        IEnumerable<T> FindAll();
        IList<T> GetAll();
    }
}
